docs/arch/adr-47.md

```diff
@@ -0,0 +1,69 @@
+# ADR 47: Highlighter Task
+
+1 Mar 2023
+
+## Context
+To support OCR verification (the DigiLeap project), with subjects that consist of an image and OCR output as text file, we will create a highlighter task. The highlighter task will allow volunteers to highlight text in the OCR output with a label or labels defined by researchers.
+
+The FEM highlighter task is an iteration on the [PFE highlighter task](https://github.com/zooniverse/Panoptes-Front-End/tree/master/app/classifier/tasks/highlighter).
+
+## Features
+To use the highlighter task a volunteer selects a label from the task area, highlights the related text in the text subject viewer, then selects a different label from the task area to highlight additional text in the text subject viewer or completes the task. Highlighted text can be unhighlighted.
```

Comment:
Is this possible and preferable? As noted in this [PFE PR 3458 comment](https://github.com/zooniverse/Panoptes-Front-End/issues/3458#issuecomment-279364964) having a user select a label then highlight text might not be the technically or accessibly preferable solution.

I'll try to create some storybooks to investigate.

Reply:
Noting with the PFE highlighter task a user highlights text then selects the related label.

Reply:
You’ll need some sort of interaction, eg. pressing a button, to indicate that selection is complete. 

Is it worth adding a link to the [Selection API](https://developer.mozilla.org/en-US/docs/Web/API/Selection) somewhere in this ADR? I assume that’s the API that you’ll be using.

Reply:
For what it’s worth, the interactions involved in text selection are:
1. Select an initial range of text in the document (using a pointer, touch or the keyboard.)
2. Adjust the extent of the initial range. On desktop, this can be done from the keyboard. On touch, you’d use the little handles at the beginning and end of the range. 
3. Do something with the selection eg. Press a button to highlight it.

Reply:
A concrete example of interaction design for text selection would be making text bold in a rich text editor.

Reply:
Would it be helpful to think of the task here as something like the toolbar in a word processor or rich text editor?